### DISCUSSION TITLE
Issues using LangGraph4j With Lambda and Java 21 JRE

### DISCUSSION THREAD

**Discussion by johnpiscani** [langgraph4j/langgraph4j#28] [Q&A]
Hello,

I am testing using LangGraph4j. I am able to run my code within Intellij, using Java17, but when I run it on AWS Lambda with Java 21 runtime in the settings, I keep getting this error: java.lang.ExceptionInInitializerError. I have two questions:

- Is LangGraph4j compatible with Java 21 runtime?
- Are there any workarounds for LangGraph4j to not use static classes to prevent this from happening, or may this issue be happening internally within the framework anyways?

Thank you!

**Comment by bsorrentino** [comment]
Hi @johnpiscani 

Please share the all the Stacktrace to understand the root cause.

Anyway I've tested LangGraph4j on  JDK 8, 11, 17, 22 and my use cases works

**Comment by johnpiscani** [comment]
Hi @bsorrentino,
Here is more info on the error.

```
Error:
java.util.concurrent.CompletionException: java.lang.ExceptionInInitializerError
Stack Trace: 
java.base/java.util.concurrent.CompletableFuture.reportJoin(Unknown Source), 
java.base/java.util.concurrent.CompletableFuture.join(Unknown Source), 
org.bsc.async.AsyncGenerator$1.next(AsyncGenerator.java:260), 
java.base/java.util.Iterator.forEachRemaining(Unknown Source),
java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source),
java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source),
java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source),
java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source),
java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source),
java.base/java.util.stream.ReferencePipeline.reduce(Unknown Source),
org.bsc.langgraph4j.CompiledGraph.invoke(CompiledGraph.java:246),
org.bsc.langgraph4j.CompiledGraph.invoke(CompiledGraph.java:257)...
```

**Reply by bsorrentino** [reply]
Hi @johnpiscani  one more question:

 which version are you using  the  `1.0-beta4` ?  I've checked the code and in `core-jdk8-1.0-beta4` and there aren't  explicit static initialiser, could be one from [Lombok](https://projectlombok.org) but, please, check also you code about that

**Reply by johnpiscani** [reply]
Hi @bsorrentino,
I am using 1.0-beta2 right now. Should I upgrade to core-jdk8-1.0-beta4?
Will look into Lombok.

**Reply by bsorrentino** [reply]
@johnpiscani  yes please and let me know, so that you help me to reach a first candidate release quickly

**Reply by johnpiscani** [reply]
Hi @bsorrentino,
Sorry for the late reply. After upgrading from 1.0-beta2 to 1.0-beta4, I am no longer seeing this ExceptionInInitialization error but am seeing a new error. Now, right as I am invoking the compiled graph, I am seeing:
Error:
```java
 java.lang.RuntimeException: java.lang.UnsupportedOperationException
Stack Trace:
java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412),
java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2050),
org.bsc.async.AsyncGenerator$1.next(AsyncGenerator.java:260),
java.base/java.util.Iterator.forEachRemaining(Iterator.java:133),
java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801),
java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484),
java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474),
java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913),
java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234),
java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:558),
org.bsc.langgraph4j.CompiledGraph.invoke(CompiledGraph.java:362),
org.bsc.langgraph4j.CompiledGraph.invoke(CompiledGraph.java:373)...
```

I have logs printed in each of my three NodeAction Classes. The workflow is not reaching any of the three NodeAction Classes based on not seeing any of these logs.

Not sure if this means the ExceptionInInitialization error will still not occur though, because before I was able to get into some of the NodeAction classes previously.

Edit: I am not even able to run this locally now, through any unit test.

**Reply by bsorrentino** [reply]
Hi @johnpiscani 

The exception is not clear about which operation raise it, can you get more information on root cause ?.
 I guess you probably have tried to update state map that is "immutable"
